include/alpaka/atomic/AtomicAtomicRef.hpp

```diff
@@ -0,0 +1,231 @@
+/* Copyright 2021 Felice Pantaleo (CERN), Andrea Bocci (CERN)
+ *
+ * This file is part of alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#include <alpaka/atomic/Traits.hpp>
+#include <alpaka/core/BoostPredef.hpp>
+
+#include <boost/atomic.hpp>
+
+#include <array>
+#include <type_traits>
+
+namespace alpaka
+{
+    //! The CPU threads accelerator atomic ops based on atomic_ref.
```

Comment:
Isn't this also for other accelerators (TBB)?

Reply:
In principle it can be for any CPU-based accelerator (threads, fibers, OpenMP 2); we have tested it only for the serial and TBB ones.
Maybe we should amend the comment to
```suggestion
    //! The CPU accelerator atomic ops based on atomic_ref.
```
(here and in `include/alpaka/atomic/AtomicStdLibLock.hpp`)

Reply:
Yes, that would be more fitting.

Reply:
This still needs to be renamed.